test/modules/EdgeHubRestartTester/src/DirectMethodEdgeHubConnector.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace EdgeHubRestartTester
+{
+    using System;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class DirectMethodEdgeHubConnector : IEdgeHubConnector
+    {
+        long directMethodCount = 0;
+        ModuleClient dmModuleClient;
+        Guid batchId;
+        DateTime runExpirationTime;
+        CancellationToken cancellationToken;
+        DateTime edgeHubRestartedTime;
+        uint restartSequenceNumber;
+        ILogger logger;
+
+        public DirectMethodEdgeHubConnector(
+            ModuleClient dmModuleClient,
+            Guid batchId,
+            DateTime runExpirationTime,
+            DateTime edgeHubRestartedTime,
+            uint restartSequenceNumber,
+            ILogger logger,
+            CancellationToken cancellationToken)
+        {
+            this.dmModuleClient = Preconditions.CheckNotNull(dmModuleClient, nameof(dmModuleClient));
+            this.batchId = batchId;
+            this.runExpirationTime = runExpirationTime;
+            this.cancellationToken = Preconditions.CheckNotNull(cancellationToken, nameof(cancellationToken));
+            this.edgeHubRestartedTime = edgeHubRestartedTime;
+            this.restartSequenceNumber = restartSequenceNumber;
+            this.logger = Preconditions.CheckNotNull(logger, nameof(logger));
+        }
+
+        public async Task StartAsync()
+        {
+            (DateTime dmCompletedTime, HttpStatusCode dmStatusCode) = await this.SendDirectMethodAsync(
+                Settings.Current.DeviceId,
+                Settings.Current.DirectMethodTargetModuleId,
+                this.dmModuleClient,
+                Settings.Current.DirectMethodName,
+                this.runExpirationTime,
+                this.cancellationToken,
+                this.logger).ConfigureAwait(false);
```

Comment:
Don't add ConfigureAwait(false). Should not be needed, and we don't use it in our code.

Reply:
Please correct me if I'm wrong. Currently the code is doing
```
main() 
{
    SendDirectMethod()
    SendMessage()
}
```

If I don't add `ConfigureAwait(false)`, the main thread will do the `SendDirectMethod()` then do the  `SendMessage()` instead of doing both in parallel which is the current behavior.

Reply:
This will also run sequentially. If you want to run them in parallel, you need to do the following - 
Task sendDMTask = this.SendDirectMethod()
Task sendMessageTask = this.SendMessage();
await Task.WhenAll(sendDMTask, sendMessageTask);

Reply:
The program.cs is already doing the `sendMessageTask = new MessageEdgeHubConnector()`
Per this article: https://devblogs.microsoft.com/dotnet/configureawait-faq/ 
It seems to suggest that `ConfigureAwait(false)` is what I want.

Reply:
Actually, you're right it doesn't do anything this particular context.  Removing the ConfigureAwait(false)